### PR TITLE
[feat][task] Add OpenMLDB task plugin

### LINKED ISSUE #9950: [Feature][Task-plugin] Add OpenMLDB task-plugin
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

I want to add the OpenMLDB task-plugin to support run cmds on OpenMLDB. 
There're some ways to impl it.
1. If we impl it like zeppelin, we'll add openmldb jdbc dependency. openmldb jdbc will use openmldb-native jar which has c++ shared lib in it. The released package will contain it. [openmldb-native jar is 156.9MB](https://mvnrepository.com/artifact/com.4paradigm.openmldb/openmldb-native/0.5.0-allinone), a little big.
2. If we impl it like datax or python task, we can let users to install openmldb python sdk later, won't add it in releases. And python wheel is smaller. [23MB](https://pypi.org/project/openmldb/#files)

Plz help me decide on which way to impl the task.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vagetablechicken**
## Purpose of the pull request

Add the task OpenMLDB, closes #9950 

## Brief change log

- Add openmldb task in dolphinscheduler-task-plugin
- Add ui 
- Add task docs(en&zh)

## Verify this pull request

This change added tests and can be verified as follows:
  - *Execute command test in module*
  - *Manually verified the change by testing locally.*

### CODE REVIEW

File: dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/pom.xml
Comment (kezhenxu94) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/pom.xml]: Hi, this effects developers' IDE installation, please open a discussion before adding this, (though I like to use this too)

Comment (vagetablechicken) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/pom.xml]: Actually I can remove this dependency in this pr. Maybe we could discuss it later?

Comment (kezhenxu94) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/pom.xml]: > Actually I can remove this dependency in this pr. Maybe we could discuss it later?

Sure

File: docs/docs/en/guide/task/openmldb.md
Comment (zhongjiajie) [inline docs/docs/en/guide/task/openmldb.md]: I think we should better use the keyword `Zookeeper` in both document and UI

File: docs/docs/en/guide/task/openmldb.md
Comment (zhongjiajie) [inline docs/docs/en/guide/task/openmldb.md]:
```suggestion
Make sure you have installed OpenMLDB Python SDK, using `pip install openmldb`.
```

File: dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/main/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTask.java
Comment (zhongjiajie) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/main/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTask.java]: Found out this task class have many similar to `PythonTask`, could we extends from python task instead of abstract?

File: dolphinscheduler-task-plugin/dolphinscheduler-task-python/src/test/java/org/apache/dolphinscheduler/plugin/task/python/PythonTaskTest.java
Comment (zhongjiajie) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-python/src/test/java/org/apache/dolphinscheduler/plugin/task/python/PythonTaskTest.java]: I think it is unrelated change here

File: dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/main/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTask.java
Comment (vagetablechicken) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/main/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTask.java]: But still have many differences. If we extend `PythonTask`, it'll only reduce ~ 50 lines.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/main/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTask.java
Comment (kezhenxu94) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/main/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTask.java]: Hi, I don't think you can simply replace all line breaks with spaces, if there are comments in the sql scripts, you will totally change the logic of it, for example, if the original sql is 

```sql
select * from users
-- some comment
inner join order on users.order_id = order.id;
```

will become 

```sql
select * from users -- some comment inner join order on users.order_id = order.id;
```

File: docs/docs/zh/guide/task/openmldb.md
Comment (jieguangzhou) [inline docs/docs/zh/guide/task/openmldb.md]:
```suggestion
# OpenMLDB 节点
```

A small mistake

**Comment by songjianet** [comment]
Please associate the corresponding issue. @vagetablechicken

File: dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/main/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTask.java
Comment (vagetablechicken) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/main/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTask.java]: Yes, I should replace \\r\\n to '\\n' in string, not the space. Comments and '\n' can be solved by sql engine. I'll fix it and add ut

**Comment by vagetablechicken** [comment]
> Please associate the corresponding issue. @vagetablechicken

done

File: docs/docs/zh/guide/task/openmldb.md
Comment (jieguangzhou) [inline docs/docs/zh/guide/task/openmldb.md]:
```suggestion
```

File: dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/main/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTask.java
Comment (vagetablechicken) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/main/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTask.java]: After the fix in `buildPythonScriptContent`, it's more similar. I'll try to extend from `PythonTask`.

**Comment by vagetablechicken** [comment]
Hi there, I've fixed all. Plz help me to check again, thx

File: docs/configs/docsdev.js
Comment (zhongjiajie) [inline docs/configs/docsdev.js:167]: You have to add zh docs into https://github.com/apache/dolphinscheduler/blob/dev/docs/configs/docsdev.js#L528 you could search `MLflow` in file https://github.com/apache/dolphinscheduler/blob/dev/docs/configs/docsdev.js and there are four match item here

File: docs/docs/en/guide/task/openmldb.md
Comment (zhongjiajie) [inline docs/docs/en/guide/task/openmldb.md]:
```suggestion
Make sure you have installed OpenMLDB Python SDK in the host where the worker server running, using `pip install openmldb`.
```

Comment (zhongjiajie) [reply docs/docs/en/guide/task/openmldb.md]: we should mention users install deps in the worker server to run the task. Same as ZH doc

File: docs/docs/zh/guide/task/openmldb.md
Comment (zhongjiajie) [inline docs/docs/zh/guide/task/openmldb.md]:
```suggestion
```

Comment (zhongjiajie) [reply docs/docs/zh/guide/task/openmldb.md]: remove unused adding

File: dolphinscheduler-ui/src/locales/modules/en_US.ts
Comment (zhongjiajie) [inline dolphinscheduler-ui/src/locales/modules/en_US.ts]: we should modify system screenshots at the same time when we change the keyword from `zk` to `zookeeper`

**Review by zhongjiajie — COMMENTED** [review]
backend and docs LGTM

**Comment by zhongjiajie** [comment]
ping @songjianet and @labbomb to take a looking the frontend code if you have time

File: dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/test/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTaskTest.java
Comment (kezhenxu94) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/test/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTaskTest.java:68]: 👍 thanks for adding this test case

**Review by kezhenxu94 — DISMISSED** [review]

File: dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/test/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTaskTest.java
Comment (vagetablechicken) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-openmldb/src/test/java/org/apache/dolphinscheduler/plugin/task/openmldb/OpenmldbTaskTest.java:68]: Got conflicts, so I'v merged dev branch. Sorry to dismiss your review.

**Review by davidzollo — DISMISSED** [review]
+1
LGTM

**Comment by zhongjiajie** [comment]
> ping @songjianet and @labbomb to take a looking the frontend code if you have time

ping @songjianet @labbomb again, PTAL the front end part if you have time, others LGTM

File: dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-openmldb.ts
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-openmldb.ts:51]: Here you should set the default value of timeoutNotifyStrategy to ['WARN'].

File: dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-openmldb.ts
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-openmldb.ts:87]: The code should be formatted.

Comment (songjianet) [reply dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-openmldb.ts:87]: For the formatting issue, you can leave it alone, I will do the formatting later. @vagetablechicken

Comment (vagetablechicken) [reply dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-openmldb.ts:87]: ok, thx

**Review by Amy0104 — APPROVED** [review]
LGTM.

**Comment by zhongjiajie** [comment]
Merging... Thank @vagetablechicken

**Pull request merged**
